underdocs-renderer/src/main/kotlin/underdocs/renderer/output/html/render/page/PageRenderer.kt

```diff
@@ -107,39 +113,45 @@ class PageRenderer(
+                                                .concat(Array.from(document.getElementsByClassName('katex-block')));
+                                            
+                                            console.log(mathElements.length)
+    
+                                            mathElements.forEach(element => {
+                                                katex.render(element.textContent, element, { 
+                                                    throwOnError: false,
+                                                    displayMode: element.classList.contains('katex-block'), 
                                                 });
-                                               })
-                                        })();
+                                            });
+                                           })
+                                    })();
                                 """.trimIndent()
                             )
                         ),
                     script()
                         .with(
                             UnescapedText(
                                 """
-                                          (function setNightModeButton() {
-                                            const nightModeButton = document.querySelector("#night-mode-button");
-                                    
-                                            if (isNightModePreferred()) {
-                                              nightModeButton.checked = true;
-                                            }
-                                            
-                                            nightModeButton.addEventListener('change', function () {
-                                              document.body.classList.toggle('night');
-                                            })
-                                          })();
+                                    (function setNightModeButton() {
+                                        const nightModeButton = document.querySelector("#night-mode-button");
+                                        
+                                        if (getPreferredTheme() === 'night') {
+                                          nightModeButton.checked = true;
+                                        }
+                                        
+                                        nightModeButton.addEventListener('change', function () {
+                                          document.body.classList.toggle('night');
+                                          
+                                          if (nightModeButton.checked === false) {
+                                            localStorage.setItem('underdocs_theme', 'day');
+                                          } else {
+                                            localStorage.setItem('underdocs_theme', 'night');
+                                          }
```

Comment:
Same as above, instead of this I could just use one-line (and leave out the if-statement):

localStorage.setItem('underdocs_theme', nightModeButton.checked ? 'night' : 'day').

Is it better this way? Maybe it's not as unreadable as the other comment above.

Reply:
Here, I would definitely go with the one-liner version. Or, if you want to be very explicit, you may consider something like this:

~~~~JavaScript
const themeChoice = nightModeButton.checked ? 'night' : 'day'

localStorage.setItem('underdocs_theme', themeChoice)
~~~~

Reply:
True, was thinking about the explicit variable version too.